tfhe/src/high_level_api/integers/types/base.rs

```diff
@@ -224,13 +225,15 @@ where
     }
 }
 
-impl<P> GenericInteger<P>
+impl<P> FheMax<&Self> for GenericInteger<P>
```

Comment:
They are used to achieve function/method 'overloading', so that we can have a.max(b) where b is either another encrypted value or a clear value like a u8.

Without it, we would need to have different methods like in integer (non scalar vs scalar), I do this in the hlapi as I think it makes the API easier/cleaner from a user perspective.

All these traits are imported in the prelude which we show the user to do `use tfhe::prelude::*;` So they should not have to worry about them (unless they use them as trait bounds in generic code)

Reply:
perfect ! thanks for the detailed explanation